docs/tutorial.md

```diff
@@ -0,0 +1,183 @@
+
+
+```sh
+mkdir -p $HOME/projects/memcached-operator
+cd $HOME/projects/memcached-operator
+# we'll use a domain of example.com
+# so all API groups will be <group>.example.com
+operator-sdk init --domain example.com --repo github.com/example/memcached-operator
+```
+
+### Create a new API and Controller 
+
+```sh
+$ operator-sdk create api --group cache --version v1alpha1 --kind Memcached --resource --controller
+Writing scaffold for you to edit...
+api/v1alpha1/memcached_types.go
+controllers/memcached_controller.go
+...
+```
+
+Define the API for the Memcached Custom Resource(CR) by modifying the Go type definitions at `api/v1alpha1/memcached_types.go` to have the following spec and status:
+
+```
+// MemcachedSpec defines the desired state of Memcached
+type MemcachedSpec struct {
+	//+kubebuilder:validation:Minimum=0
+	// Size is the size of the memcached deployment
+	Size int32 `json:"size"`
+}
+
+// MemcachedStatus defines the observed state of Memcached
+type MemcachedStatus struct {
+	// Nodes are the names of the memcached pods
+	Nodes []string `json:"nodes"`
+}
+```
+
+### Now create docker image and push to the dockerhub
+
+Change below code in Makefile
+
+```
+-IMG ?= controller:latest
++IMG ?= $(IMAGE_TAG_BASE):$(VERSION)
+```
+
+Once done, you do not have to set IMG or any other image variable in the CLI. The following command will build and push an operator image tagged as example.com/memcached-operator:v0.0.1 to Docker Hub:
+
+```
+make docker-build docker-push
+```
+
+### Generating CRD manifests 
+
+```sh
+$ kustomize build config/default > manifest/manifest.yaml
+```
+
+Need to update `manifest.yaml` file.
+
+Update image tag from `controller:latest` to `example.com/memcached-operator:v0.0.1`
+
+Then, remove `ClusterRole` with name `memcached-operator-manager-role` and `ClsuerRoleBinding` with name `memcached-operator-manager-rolebinding`.
+
+Now, create the `ScopeTemplate` and add above removed `ClsuterRole` in spec in `manifests/scopeTemplate.yaml`.
+
+```yaml
+apiVersion: operators.io.operator-framework/v1alpha1
+kind: ScopeTemplate
+metadata:
+  name: scopetemplate-sample-manager
+spec:
+  clusterRoles:
+  - generateName: manager-role
+    rules:
+    - apiGroups:
+      - "*"
+      resources:
+      - "*"
+      verbs:
+      - create
+      - delete
+      - get
+      - list
+      - patch
+      - update
+      - watch
+    - apiGroups:
+      - cache.example.com
+      resources:
+      - memcacheds/finalizers
+      verbs:
+      - update
+    - apiGroups:
+      - cache.example.com
+      resources:
+      - memcacheds/status
+      verbs:
+      - get
+      - patch
+      - update
+    subjects:
+    - kind: ServiceAccount
+      name: memcached-operator-controller-manager
+      namespace: memcached-operator-system
+```
+
+Before this install `ScopeTemplate` CRD from `oria-operator`.
+
+```
+$ make install
```

Comment:
Where are they running this `make install` command? Won't that install the memcached CRDs? I presume I'm still in the memcached-operator directory

Reply:
Yeah, User need to diver to `oria-operator` and do this as we are running `ScopeTemplate` from manifests.